Add voiceservices sdk

sebastianrex opened the pull request:
This PR adds the Azure.ResourceManager.VoiceServices SDK for the "Microsoft.VoiceServices" ARM api defined here: https://github.com/Azure/azure-rest-api-specs/tree/main/specification/voiceservices/resource-manager/Microsoft.VoiceServices/stable/2023-01-31

I've followed the guidance here: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=sdk-generation-from-scratch to create this PR.

The .NET namespace name has been approved in this issue: https://github.com/Azure/azure-sdk-pr/issues/842

I've used the `EdgeOrder` tests as a starting point for how to write the tests, as per the Wiki instructions. I think the tests I've written are sufficient.

These changes were previous reviewed within our team in a PR to our branch in the private repo: https://github.com/Azure/azure-sdk-for-net-pr/pull/1802, but since then I've regenerated it with the correct name and updated it to the latest stable API version.

The tests all pass locally. I've noticed that CI is failing, complaining that it doesn't have the `VOICESERVICES_TENANT_ID` or `TENANT_ID` env vars, but surely these aren't expected to be checked in? Is this CI failure expected in a PR off a fork? If not, what should I do to fix it up?

The ability to use the Microsoft.VoiceServices API is hidden behind AFEC flags, and we expect this to continue. So I'm not sure how I square this with the CI - do the .NET SDK team have a single subscription that CI uses? If so, I'll need to add the required AFEC flag for that subscription.

Fixes https://github.com/Azure/azure-sdk-for-net/issues/34367

sebastianrex committed:
Add voiceservices SDK folders

sebastianrex committed:
Add generated code

sebastianrex committed:
Update mgmt CI

sebastianrex committed:
Export API

sebastianrex committed:
Add ci.yml

sebastianrex committed:
Try including test framework

sebastianrex committed:
Working tests build

sebastianrex committed:
Add tests

sebastianrex committed:
Add test-resources.json script

sebastianrex committed:
Add SessionRecords

archerzz committed:
fix test mode

sebastianrex commented:
Hi @ArcturusZhang, we've hoping to get this released by the end of Februrary to meet our APEX commitment. Do you know when you'll be able to review this?

ArcturusZhang commented on sdk/voiceservices/Azure.ResourceManager.VoiceServices/README.md:
```diff
@@ -0,0 +1,80 @@
+# Microsoft Azure VoiceServices management client library for .NET
+
+**[Describe the service briefly first.]**
```
Could we add some description of this RP here?

ArcturusZhang commented on sdk/voiceservices/Azure.ResourceManager.VoiceServices/src/autorest.md:
```diff
@@ -0,0 +1,49 @@
+skip-csproj: true
+modelerfour:
+  flatten-payloads: false
+tag: package-2023-01-31
```
since the default tag in the above readme is exactly this tag, could we just remove this line?
This helps our SDK automation pipeline in the swagger review process to keep the SDK automatically up-to-date and we could preview the SDK changes on a swagger PR

ArcturusZhang commented on sdk/voiceservices/test-resources.json:
```diff
@@ -0,0 +1,6 @@
+{
+  "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+  "contentVersion": "1.0.0.0",
+  "resources": [
+  ]
+}
```
As far as I know, only data plane packages need this file. Could we remove this?

sebastianrex replied:
Sure. I added it because the initial setup required it (without it I got an error running the script that sets up the KV etc. mentioned in the docs), so I assumed that all SDK packages needed it. But it looks like the tests now pass without it again, so I'll remove it.

sebastianrex committed:
Markups

sebastianrex committed:
Update CHANGELOG

sebastianrex commented:
Thanks @ArcturusZhang , I've addressed all 3 of your suggestions now.

I've also updated the CHANGELOG in preparation for the SDK release, per the release instructions. I figured now that it has been reviewed I may as well perform the release changelog update in the same PR

ArthurMa1978 commented:
Hi @sebastianrex , the Sdk can't be released without an approval of namespace review (like this https://github.com/Azure/azure-sdk/issues/5255), please establish a review for your service?

ArthurMa1978 reviewed (dismissed):
Need namespace review

ArthurMa1978 commented:
/azp run prepare-pipelines

ArthurMa1978 commented on sdk/voiceservices/ci.yml:
```diff
@@ -0,0 +1,32 @@
+# NOTE: Please refer to https://aka.ms/azsdk/engsys/ci-yaml before editing this file.
+
+trigger:
```
refer https://github.com/Azure/azure-sdk-for-net/blob/main/sdk/edgeorder/ci.mgmt.yml to fix this file for ci pipeline, and rename this file to `ci.mgmt.yml`

sebastianrex replied:
Ok, fixed.

For reference, I was following the docs here: https://dev.azure.com/azure-sdk/internal/_wiki/wikis/internal.wiki/78/Mgmt-.NET-SDK-release-process?anchor=sdk-generation-from-scratch, which have:

> Create a ci.yml file under the root folder of this SDK which will be used to create the CI pipeline. Here the folder is root\sdk\servicelinker, and the content in this file is:

(and then an example file). Is it worth someone correcting those docs?

sebastianrex committed:
Rename ci.yml to ci.mgmt.yml and fix

ArthurMa1978 commented on sdk/voiceservices/Azure.ResourceManager.VoiceServices/CHANGELOG.md:
```diff
@@ -0,0 +1,19 @@
+# Release History
+
+## 1.0.0 (2023-02-16)
```
Change the version to `1.0.0-beta.1` as any new stable should have beta ahead for 1 month

sebastianrex replied:
Ok, I wasn't sure because our service is already GA, but I've now corrected, sorry about that

ArthurMa1978 commented on sdk/voiceservices/Azure.ResourceManager.VoiceServices/src/Properties/AssemblyInfo.cs:
```diff
@@ -0,0 +1,9 @@
+
+using System.Runtime.CompilerServices;
+
+[assembly: Azure.Core.AzureResourceProviderNamespace("VoiceServices")]
```
The value should be `Microsoft.VoiceServices`

sebastianrex replied:
Thanks, I've fixed it

ArthurMa1978 commented:
/azp run prepare-pipelines

ArthurMa1978 commented:
/azp run net - voiceservices - mgmt

sebastianrex committed:
Fix up AssemblyInfo namespace

sebastianrex committed:
Update CHANGELOG

azure-sdk commented:
**API change check**

APIView has identified API level changes in this PR and created following API reviews.

[Azure.ResourceManager.VoiceServices](https://apiview.dev/Assemblies/Review/fca2b1d08fd04928b50f4a6e368e4056)

sebastianrex committed:
Rerecord tests

sebastianrex committed:
Fix spellcheck

sebastianrex commented:
Sorry, I was told to rebase off `main` to fix up the recorded tests, and doing so lost the commit you added in the meantime. I've reproduced it as the last commit on this branch

ArthurMa1978 commented:
> Sorry, I was told to rebase off `main` to fix up the recorded tests, and doing so lost the commit you added in the meantime. I've reproduced it as the last commit on this branch

please rerun `dotnet build /t:GenerateTest` to update samples (`dotnet build /t:GenerateTests` if you synced up with main).

sebastianrex committed:
Regenerate code after rebase off main

sebastianrex commented:
> > Sorry, I was told to rebase off `main` to fix up the recorded tests, and doing so lost the commit you added in the meantime. I've reproduced it as the last commit on this branch
> 
> please rerun `dotnet build /t:GenerateTest` to update samples (`dotnet build /t:GenerateTests` if you synced up with main).

Ah, I didn't realise that would make a difference, I've now updated the samples

sebastianrex committed:
Regenerate snippets

ArthurMa1978 committed:
update
